scipy/interpolate/_fitpack_repro.py

```diff
@@ -300,8 +373,9 @@ def _generate_knots_impl(x, y, *, w=None, xb=None, xe=None, k=3, s=0, nest=None)
 
             # recompute if needed
             if j < nplus - 1:
-                residuals = _get_residuals(x, y, t, k, w=w)
+                residuals, _ = _get_residuals(x, y, t, k, w=w, periodic=periodic)
 
+        fpold = fp
```

Comment:
Please check if this is still needed (same is done at the beginning of the loop)